Description: Fix find_package(ITK) with only required COMPONENTS list
Author: Rashad Kanavath <rashad.kanavath@c-s.fr>
Last-Update: 2015-12-22

diff --git a/Modules/ThirdParty/ITK/CMakeLists.txt b/Modules/ThirdParty/ITK/CMakeLists.txt
index 55b0eb8..8b41866 100644
--- a/Modules/ThirdParty/ITK/CMakeLists.txt
+++ b/Modules/ThirdParty/ITK/CMakeLists.txt
@@ -2,7 +2,80 @@ project(OTBITK)

 set(OTBITK_SYSTEM_INCLUDE_DIRS ${ITK_INCLUDE_DIRS})
 set(OTBITK_LIBRARIES "${ITK_LIBRARIES}")
-set(OTBITK_EXPORT_CODE_BUILD "find_package(ITK REQUIRED HINTS ${ITK_DIR})")
+set(OTBITK_ITK_REQUIRED_MODULE_LIST
+    ITKCommon
+    ITKFiniteDifference
+    ITKGPUCommon
+    ITKGPUFiniteDifference
+    ITKImageAdaptors
+    ITKImageFunction
+    ITKMesh
+    ITKQuadEdgeMesh
+    ITKSpatialObjects
+    ITKTransform
+    ITKAnisotropicSmoothing
+    ITKAntiAlias
+    ITKBiasCorrection
+    ITKBinaryMathematicalMorphology
+    ITKColormap
+    ITKConvolution
+    ITKCurvatureFlow
+    ITKDeconvolution
+    ITKDenoising
+    ITKDisplacementField
+    ITKDistanceMap
+    ITKFastMarching
+    ITKFFT
+    ITKGPUAnisotropicSmoothing
+    ITKGPUImageFilterBase
+    ITKGPUSmoothing
+    ITKGPUThresholding
+    ITKImageCompare
+    ITKImageCompose
+    ITKImageFeature
+    ITKImageFilterBase
+    ITKImageFusion
+    ITKImageGradient
+    ITKImageGrid
+    ITKImageIntensity
+    ITKImageLabel
+    ITKImageSources
+    ITKImageStatistics
+    ITKLabelMap
+    ITKMathematicalMorphology
+    ITKPath
+    ITKQuadEdgeMeshFiltering
+    ITKSmoothing
+    ITKSpatialFunction
+    ITKThresholding
+    ITKEigen
+    ITKNarrowBand
+    ITKNeuralNetworks
+    ITKOptimizers
+    ITKOptimizersv4
+    ITKPolynomials
+    ITKStatistics
+    ITKRegistrationCommon
+    ITKGPURegistrationCommon
+    ITKGPUPDEDeformableRegistration
+    ITKMetricsv4
+    ITKPDEDeformableRegistration
+    ITKRegistrationMethodsv4
+    ITKClassifiers
+    ITKConnectedComponents
+    ITKDeformableMesh
+    ITKKLMRegionGrowing
+    ITKLabelVoting
+    ITKLevelSets
+    ITKLevelSetsv4
+    ITKMarkovRandomFieldsClassifiers
+    ITKRegionGrowing
+    ITKSignedDistanceFunction
+    ITKVoronoi
+    ITKWatersheds)
+
+set(OTBITK_EXPORT_CODE_BUILD "find_package(ITK REQUIRED COMPONENTS ${OTBITK_ITK_REQUIRED_MODULE_LIST} HINTS ${ITK_DIR})")
+
 set(OTBITK_EXPORT_CODE_INSTALL ${OTBITK_EXPORT_CODE_BUILD})

 otb_module_impl()
